commons/src/main/java/io/aiven/kafka/tieredstorage/commons/ObjectKey.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons;
+
+import java.text.NumberFormat;
+import java.util.Objects;
+
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentId;
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentMetadata;
+
+public final class ObjectKey {
+    public enum Suffix {
+        LOG("log"),
+        OFFSET_INDEX("index"),
+        TIME_INDEX("timeindex"),
+        PRODUCER_SNAPSHOT("snapshot"),
+        TXN_INDEX("txnindex"),
+        LEADER_EPOCH_CHECKPOINT("leader-epoch-checkpoint"),
+        MANIFEST("rsm-manifest");
+
+        public final String value;
+        Suffix(final String value) {
+            this.value = value;
+        }
+    }
+
+    public static String key(final RemoteLogSegmentMetadata remoteLogSegmentMetadata,
+                             final Suffix suffix) {
```

Comment:
And what about prefix?

Reply:
I'm thinking if it should be a part of the S3 implementation config and not be exposed to this level...

Reply:
I still don't know, though. WDYT guys @AnatolyPopov @jeqo? Basically:
- the prefix is an storage implementation-specific setting, like the bucket; or
- the prefix is universal and should be used for all the storage implementations.

Reply:
I agree, this is a static method and we can provide whatever here when we will use it, so just providing empty fi it's not needed is fine by me. On the other hand if we will not provide it here then we will need to do something on top of this to add a prefix in S3 implementation at least.

Reply:
Ok, added the prefix

Reply:
Should the `/` after prefix be added here to avoid keeping in mind that it should be added? The reason is that we sould most probably like to put all the stuff in `sub-directory` in S3 for example

Reply:
Just a though: what if users don't want to make it a directory and just adding a cluster name as prefix to the topic names? e.g. `cluster-1-t1-AAAAA1/...`
We could just document that it's up to them to define whether to make the prefix a sub-directory or not. wdyt?

Reply:
Perhaps you are right, was thinking from a bit different perspective so nm.

Reply:
> Just a though: what if users don't want to make it a directory and just adding a cluster name as prefix to the topic names?

Yeah, that's the idea behind not adding `/`. Since S3 and other object storages don't treat `/` anyhow special, I think it's fair to avoid this on our side too (despite we're doing some `/` ourselves, but that's unavoidable).

> We could just document that it's up to them to define whether to make the prefix a sub-directory or not.

Yeah. After the changes here we'll need to update the RSM config class, so I made a ticket: https://github.com/aiven/tiered-storage-for-apache-kafka/issues/168